backend/main/server/app.go

```diff
@@ -1875,3 +1865,21 @@ func (a *App) validateTimestamp(timestamp string, expiry int) error {
 	}
 	return nil
 }
+
+func (a *App) processSnapshotStatus(s *models.Strategy, p *models.Proposal) error {
+	var processing = "processing"
+	if s.Contract.Name != nil && p.Snapshot_status == &processing {
+		snapshotResponse, err := a.SnapshotClient.GetSnapshotStatus(s.Contract)
```

Comment:
Get snapshot at specific blockheight

Reply:
renamed to `GetSnapshotStatusAtBlockHeight` and updated logic. Room for improvement in `setSnapshotUrl` but I have made a separate ticket for it